docker/distribution_error.go

```diff
@@ -0,0 +1,148 @@
+// Code below is taken from https://github.com/distribution/distribution/blob/a4d9db5a884b70be0c96dd6a7a9dbef4f2798c51/registry/client/errors.go
+// Copyright 2022 github.com/distribution/distribution authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+	dockerChallange "github.com/docker/distribution/registry/client/auth/challenge"
```

Comment:
```suggestion
	dockerChallenge "github.com/docker/distribution/registry/client/auth/challenge"
```
(or just leave it as `challenge` without renaming?)

Reply:
oh yeah lets not rename imports if it is not required

Reply:
looks like we have to rename since challenge is already declared here: https://github.com/containers/image/blob/3aed337418462460c111a3beade6723832af5be0/docker/wwwauthenticate.go#L13
but will fix the typo.